quisp/modules/ConnectionManager.cc

```diff
@@ -653,68 +595,72 @@ void ConnectionManager::intermediate_reject_req_handler(RejectConnectionSetupReq
+  unsigned long ruleset_id = createUniqueId();
   int rule_index = 0;
-  std::vector<int> partners = {conf.left_partner, conf.right_partner};
-  RuleSet *EntanglementSwapping = new RuleSet(RuleSet_id, owner, partners);
-  Rule *SwappingRule = new Rule(RuleSet_id, rule_index);
-  EntanglementSwapping->setRule_ptr(SwappingRule);
-  Condition *Swap_condition = new Condition();
+
   Clause *resource_clause_left = new EnoughResourceClauseLeft(conf.left_partner, conf.lres);
   Clause *resource_clause_right = new EnoughResourceClauseRight(conf.right_partner, conf.rres);
-  Swap_condition->addClause(resource_clause_left);
-  Swap_condition->addClause(resource_clause_right);
-  SwappingRule->setCondition(Swap_condition);
-  quisp::rules::Action *swap_action = new SwappingAction(RuleSet_id, rule_index, conf.left_partner, conf.lqnic_type, conf.lqnic_index, conf.lqnic_address, conf.lres,
-                                                         conf.right_partner, conf.rqnic_type, conf.rqnic_index, conf.rqnic_address, conf.rres, conf.self_left_qnic_index,
-                                                         conf.self_left_qnic_type, conf.self_right_qnic_index, conf.self_right_qnic_type);
-  SwappingRule->setAction(swap_action);
-  EntanglementSwapping->addRule(SwappingRule);
-  rule_index++;
-
-  return EntanglementSwapping;
+
+  Condition *condition = new Condition();
+  condition->addClause(resource_clause_left);
+  condition->addClause(resource_clause_right);
+
+  quisp::rules::Action *action = new SwappingAction(ruleset_id, rule_index, conf.left_partner, conf.lqnic_type, conf.lqnic_index, conf.lqnic_address, conf.lres, conf.right_partner,
+                                                    conf.rqnic_type, conf.rqnic_index, conf.rqnic_address, conf.rres, conf.self_left_qnic_index, conf.self_left_qnic_type,
+                                                    conf.self_right_qnic_index, conf.self_right_qnic_type);
+
+  Rule *rule = new Rule(ruleset_id, rule_index);
+  rule->setCondition(condition);
+  rule->setAction(action);
+
+  std::vector<int> partners = {conf.left_partner, conf.right_partner};
+
+  RuleSet *ruleset = new RuleSet(ruleset_id, owner, partners);
+  ruleset->addRule(rule);
+  ruleset->setRule_ptr(rule);
+
+  return ruleset;
 }
 
-RuleSet *ConnectionManager::generateTomographyRuleSet(unsigned long RuleSet_id, int owner, int partner, int num_of_measure, QNIC_type qnic_type, int qnic_index,
-                                                      int num_resources) {
-  // tomography ruleset
-  int rule_index = 0;
-  RuleSet *tomography = new RuleSet(RuleSet_id, owner, partner);
-  Rule *Random_measure_tomography = new Rule(RuleSet_id, rule_index);
+RuleSet *ConnectionManager::generateTomographyRuleSet(int owner, int partner, int num_of_measure, QNIC_type qnic_type, int qnic_index, int num_resources) {
+  unsigned long ruleset_id = createUniqueId();
```

Comment:
if we use an unique id as `ruleset_id`, we don't need to pass it as the parameter. 
so I just create an unique id inside the method.